src/YawPitchControl/input/FusionPoseSensor.js

```diff
@@ -0,0 +1,239 @@
+
+const K_FILTER = 0.98;
+const PREDICTION_TIME_S = 0.040;
+
+export default class FusionPoseSensor extends Component {
+	constructor() {
+		super();
+
+		this.deviceMotion = new DeviceMotion();
+
+		this.accelerometer = new MathUtil.Vector3();
+		this.gyroscope = new MathUtil.Vector3();
+
+		this._onDeviceMotionChange = this._onDeviceMotionChange.bind(this);
+		this._onScreenOrientationChange = this._onScreenOrientationChange.bind(this);
+
+		this.filter = new ComplementaryFilter(K_FILTER);
+		this.posePredictor = new PosePredictor(PREDICTION_TIME_S);
+
+		this.filterToWorldQ = new MathUtil.Quaternion();
+
+		this.isFirefoxAndroid = Util.isFirefoxAndroid();
+		this.isIOS = Util.isIOS();
+
+		// Ref https://github.com/immersive-web/cardboard-vr-display/issues/18
+		this.isChromeUsingDegrees = CHROME_VERSION >= 66;
+
+		this._isEnabled = false;
+
+		// Set the filter to world transform, depending on OS.
+		if (this.isIOS) {
+			this.filterToWorldQ.setFromAxisAngle(new MathUtil.Vector3(1, 0, 0), Math.PI / 2);
+		} else {
+			this.filterToWorldQ.setFromAxisAngle(new MathUtil.Vector3(1, 0, 0), -Math.PI / 2);
+		}
+
+		this.inverseWorldToScreenQ = new MathUtil.Quaternion();
+		this.worldToScreenQ = new MathUtil.Quaternion();
+		this.originalPoseAdjustQ = new MathUtil.Quaternion();
+		this.originalPoseAdjustQ.setFromAxisAngle(new MathUtil.Vector3(0, 0, 1),
+			-window.orientation * Math.PI / 180);
+
+		this._setScreenTransform();
+		// Adjust this filter for being in landscape mode.
+		if (Util.isLandscapeMode()) {
+			this.filterToWorldQ.multiply(this.inverseWorldToScreenQ);
+		}
+
+		// Keep track of a reset transform for resetSensor.
+		this.resetQ = new MathUtil.Quaternion();
+
+		this.deviceMotion.on("devicemotion", this._onDeviceMotionChange);
+		this.enable();
+	}
+	enable() {
+		if (this.isEnabled()) {
+			return;
+		}
+		this.deviceMotion.enable();
+		this._isEnabled = true;
+		window.addEventListener("orientationchange", this._onScreenOrientationChange);
+	}
+	disable() {
+		if (!this.isEnabled()) {
+			return;
+		}
+		this.deviceMotion.disable();
+		this._isEnabled = false;
+		window.removeEventListener("orientationchange", this._onScreenOrientationChange);
+	}
+	isEnabled() {
+		return this._isEnabled;
+	}
+	destroy() {
+		this.disable();
+		this.deviceMotion = null;
+	}
+	_triggerChange() {
+		const orientation = this.getOrientation();
+
+		// if orientation is not prepared. don't trigger change event
+		if (!orientation) {
+			return;
+		}
+
+		if (!this._prevOrientation) {
+			this._prevOrientation = orientation;
+			return;
+		}
+
+		if (quat.equals(this._prevOrientation, orientation)) {
+			return;
+		}
+
+		this.trigger("change", {quaternion: orientation});
+	}
+	getOrientation() {
+		let orientation;
+
+		// Hack around using deviceorientation instead of devicemotion
+		if (this.deviceMotion.isWithoutDeviceMotion && this._deviceOrientationQ) {
+			this.deviceOrientationFixQ = this.deviceOrientationFixQ || (function() {
```

Comment:
How about arrow function?

Reply:
This might be a perfect use case for arrow function. Thanks for checking.